spinalcordtoolbox/reports/qc.py

```diff
@@ -743,107 +612,98 @@ def generate_qc(fname_in1, fname_in2=None, fname_seg=None, plane=None, angle_lin
     :param fname_in2: str: File name of input image #2
     :param fname_seg: str: File name of input segmentation
     :param plane: str: Orientation of the QC. Can be: Axial, Sagittal.
-    :param angle_line: list: Angle [in rad, wrt. vertical line, must be between -pi and pi] to apply to the line overlaid on the image, for\
-    each slice, for slice that don't have an angle to display, a nan is expected. To be used for assessing cord orientation.
     :param args: args from parent function
     :param path_qc: str: Path to save QC report
     :param dataset: str: Dataset name
     :param subject: str: Subject name
-    :param path_img: dict: Path to image to display (e.g., a graph), instead of computing the image from MRI.
     :param process: str: Name of SCT function. e.g., sct_propseg
     :param fps: float: Number of frames per second for output gif images. Used only for sct_frmi_moco and sct_dmri_moco.
     :return: None
     """
     logger.info('\n*** Generate Quality Control (QC) html report ***')
-    dpi = 300
-    qcslice_type = None
-    qcslice_operations = None
-    qcslice_layout = None
+    dpi = 300  # Output resolution of the image
+    qcslice_type: qcslice.Slice
+    action_list: List[Callable[[QcImage, np.ndarray, Axes], None]]
+    qcslice_layout: Callable[[qcslice.Slice], (List[np.ndarray] | Tuple[List[List[np.ndarray]], List[Tuple[int, int]]])]
```

Comment:
After properly reinstalling Python 3.8, I'm getting this error from PyCharm:

> Python version 3.8 does not allow writing union types as X | Y

I'm not sure if this is a false-positive ([PY-51872](https://youtrack.jetbrains.com/issue/PY-51872)) or if it's a valid concern, but I thought I'd mention it just in case.

Reply:
I don't mind changing it to use `Union[X, Y]`, especially since it's already a very ugly type. Which is a hint that this code is trickier than it should be!